src/cargo/ops/cargo_install.rs

```diff
@@ -192,6 +188,16 @@ impl<'cfg, 'a> InstallablePackage<'cfg, 'a> {
             ws.current()?.clone()
         };
 
+        // When we build this package, we want to build the *specified* package only,
+        // and avoid building e.g. workspace default-members instead. Do so by constructing
+        // specialized compile options specific to the identified package.
+        // See test `path_install_workspace_root_despite_default_members`.
+        let mut opts = original_opts.clone();
+        // Unlike install source tracking (for --git, see above), use the spec url from the
+        // build workspace (hence unconditional `ws.current()` instead of `pkg` to get `package_id()`).
```

Comment:
I think [this explanation](https://github.com/rust-lang/cargo/pull/12015#issuecomment-1518234178) is better than the current comment, especially the mention of checkout URL. Could you reword it a bit?

BTW, I forgot why we cannot use Git URL here. Is it related to some parsing ambiguity around file URLs?

Reply:
I've updated the comment, hope this one is clearer.

If we try to `cargo build -p` a git url, I believe that just always fails (even in the workspace where we cloned that repo, because now that workspace considers that package to have a different `file://` url, not the source git url.)